gt/main.c

```diff
@@ -1697,14 +1695,37 @@ gt_stage2(void *arg)
 }
 
 int
-run_gt(struct net_config *net_conf, struct gt_config *gt_conf)
+run_gt(struct net_config *net_conf, struct gt_config *gt_conf,
+	const char *lua_base_directory, const char *lua_policy_file)
 {
 	int ret, i;
 
-	if (net_conf == NULL || gt_conf == NULL) {
+	if (net_conf == NULL || gt_conf == NULL ||
+			lua_base_directory == NULL ||
+			lua_policy_file == NULL) {
+		ret = -1;
+		goto out;
+	}
+
+	gt_conf->lua_base_directory = rte_malloc(
+		"lua_base_directory", strlen(lua_base_directory) + 1, 0);
+	if (gt_conf->lua_base_directory == NULL) {
```

Comment:
Instead of repeating the code here for `gt_conf->lua_policy_file` below, create function `gk_strdup()` that works as `strdup(3)`.